### PR TITLE
[PLATFORM-1191] Disable stream edit fields when no write permissions

### PR DESCRIPTION
**Pull request by timoxley**
Updates various components used by the stream edit page to respect the `disabled` attribute.

### To Test

1. Share a stream that has some fields to tester2@streamr.com with read-only access.
2. Open stream as tester2@streamr.com
3. Note all stream form elements are now appropriately non-editable

### Updated

* DatePicker: Support disabled attribute
* Sortable stream fields: treat disabled fields as locked
* ConfigureView: Remove "Delete" DropdownAction from disabled fields
* SelectInput: Correctly mapped disabled attribute to react-select's isDisabled
* NumberField: Correctly disable incr/decr buttons when input disabled
* SecurityView: Support disabled attribute
 
### Also

* Converted all files in `src` that used windows line endings via `dos2unix`

### CODE REVIEW

File: app/src/userpages/components/StreamPage/Show/ConfigureView/index.jsx [hunk 1]
```diff
@@ -174,8 +174,8 @@ export class ConfigureView extends Component<Props, State> {
                                                     value={field.name}
                                                     onChange={(e) => this.onFieldNameChange(field.name, e.target.value)}
                                                     disabled={disabled}
-                                                    actions={[
-                                                        <DropdownActions.Item key="delete" onClick={() => this.deleteField(field.name)}>
+                                                    actions={!disabled && [
+                                                        <DropdownActions.Item disabledkey="delete" onClick={() => this.deleteField(field.name)}>
```
Comment (mondoreale) [inline app/src/userpages/components/StreamPage/Show/ConfigureView/index.jsx]: What does that `disabledkey` do?

Comment (timoxley) [reply app/src/userpages/components/StreamPage/Show/ConfigureView/index.jsx]: no idea how that got in there, it should just be `key`, will change

File: app/src/shared/components/DatePicker/index.jsx [hunk 2]
```diff
@@ -51,7 +51,7 @@ class DatePicker extends React.Component<Props> {
                 noUnderline
             >
                 {({ value, onFocusChange, setAutoCompleted, ...rest0 }: InnerProps) => (
-                    <WithCalendar {...rest0} disabled={isMobile}>
+                    <WithCalendar {...rest0} disabled={isMobile || disabled}>
```
Comment (fonty1) [inline app/src/shared/components/DatePicker/index.jsx]: `rest0` typo or?

**Comment by fonty1** [comment]
I've asked Matt for the disabled states of text inputs because I can't find them on abstract:
https://share.goabstract.com/b4286f13-008d-4a4a-bd21-0a9ad5b21a3a?mode=design&sha=017447376392be759e52215d9df7046c6e1ede70

Would also be good to get the hover icon 🚫 on those inputs.

- ColorPicker.pcss -> camelCase

- Blue borders on a disabled input: 
Screenshot 2020-01-10 at 11 10 03

Otherwise all good 👍

**Comment by timoxley** [comment]
Did another pass on the disabled styles:

image

image

**Comment by mondoreale** [comment]
@timoxley, let me rebase this one… Looks like I may know something about these conflicts.

**Comment by mondoreale** [comment]
@fonty1 @timoxley
 
- I rebased it nicely
- added 🚫cursor to all components on the stream edit page
- Tweaked disabled Meatball's styling (all colors support it now!)
- Renamed Select's `isDisabled` to `disabled` to follow naming conventions

etc. etc.

lmk if you see anything blocking here. Otherwise let's merge it ASAP (today maybe?)

Cheers,
MR

File: app/src/marketplace/components/SelectField/index.jsx [hunk 3]
```diff
@@ -26,8 +26,7 @@ export const SelectField = ({ error, isProcessing, disabled, ...inputProps }: Se
                 controlClassName={cx({
                     [styles.withError]: !!hasError,
                 })}
-                // FIXME(MR): Let's rename `isDisabled` to `disabled`, come on!
-                isDisabled={disabled}
+                disabled={disabled}
```
Comment (timoxley) [inline app/src/marketplace/components/SelectField/index.jsx]: 👍

File: app/src/shared/components/Meatball/meatball.pcss [hunk 4]
```diff
@@ -65,8 +67,8 @@ a:--enter .blue {
   fill: var(--grey2);
 }
 
-.gray:--enter,
-a:--enter .gray {
+.gray:not(.disabled):--enter,
```
Comment (timoxley) [inline app/src/shared/components/Meatball/meatball.pcss]: this is very annoying but 🤷‍♀

File: app/src/userpages/components/StreamPage/Show/InfoView/index.jsx [hunk 5]
```diff
@@ -109,6 +109,7 @@ export const InfoView = ({ disabled }: Props) => {
                     {I18n.t('userpages.streams.edit.details.streamId')}
                 </Label>
                 <ActionsDropdown
+                    disabled={disabled}
```
Comment (timoxley) [inline app/src/userpages/components/StreamPage/Show/InfoView/index.jsx]: @mondoreale Hm, read-only users should still be able to copy the stream id

File: app/src/shared/components/Ui/Numeric.jsx [hunk 6]
```diff
@@ -51,7 +51,7 @@ const Buttons = styled(UnstyledButtons)`
     }
 
     button:disabled {
-        pointer-events: none;
+        cursor: not-allowed;
```
Comment (timoxley) [inline app/src/shared/components/Ui/Numeric.jsx]: I don't know if this is good

File: app/src/userpages/components/StreamPage/Show/InfoView/index.jsx (hunk 5 above)
Comment (mondoreale) [reply app/src/userpages/components/StreamPage/Show/InfoView/index.jsx]: Oh, damn, yeah, I guess. I went full-disable with this one accidently. 😱

File: app/src/shared/components/Ui/Numeric.jsx (hunk 6 above)
Comment (mondoreale) [reply app/src/shared/components/Ui/Numeric.jsx]: What's on your mind?

**Comment by mondoreale** [comment]
@timoxley, I'm merging this PR in (wanna avoid dragging it on for another day). Let's discuss/address [the `not-allowed` thing](https://github.com/streamr-dev/streamr-platform/pull/829#discussion_r374101318) separately. Hope that works for you.

**Pull request merged**